src/main/java/taxi/dao/jdbcimpl/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,210 @@
+        }
+        return carList;
+    }
+
+    private Car createCar(ResultSet resultSet, Connection connection) {
+        try {
+            Long carId = resultSet.getObject("car_id", Long.class);
+            String carModel = resultSet.getObject("model", String.class);
+            String manufacturerName = resultSet.getObject("manufacturer_name", String.class);
+            String manufacturerCountry = resultSet.getObject("manufacturer_country", String.class);
+            Long manufacturersId = resultSet.getObject("manufacturer_id", Long.class);
+            Manufacturer manufacturer = new Manufacturer(manufacturerName, manufacturerCountry);
+            manufacturer.setId(manufacturersId);
+            Car newCar = new Car(carModel, manufacturer);
+            newCar.setDrivers(getAllDriversByCarId(carId, connection));
+            newCar.setId(carId);
+            return newCar;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a 'car', "
+                    + "can not parse data", e);
+        }
+    }
+
+    private List<Driver> getAllDriversByCarId(Long carId, Connection connection) {
+        String selectQuery = "SELECT d.id, d.driver_name, d.license_number "
+                + "FROM cars_drivers "
+                + "LEFT JOIN drivers d ON d.id = cars_drivers.driver_id "
+                + "LEFT JOIN cars c ON c.id = cars_drivers.car_id "
+                + "WHERE car_id = ? AND d.deleted = FALSE;";
+        List<Driver> driverList = new ArrayList<>();
+        try (PreparedStatement preparedStatement = connection.prepareStatement(selectQuery)) {
+            preparedStatement.setLong(1, carId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                driverList.add(createDriver(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not find drivers for car with id: " + carId, e);
+        }
+        return driverList;
+    }
+
+    private Driver createDriver(ResultSet resultSet) {
+        try {
+            Driver newDriver = new Driver(resultSet.getObject("driver_name", String.class),
+                    resultSet.getObject("license_number", String.class));
+            newDriver.setId(resultSet.getObject("id", Long.class));
+            return newDriver;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not create a 'driver', "
+                    + "can not parse data", e);
+        }
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars c SET model = ?, "
+                + "manufacturer_id = ? WHERE c.id = ? AND c.deleted = FALSE";
+        String unlinkQuery = "DELETE FROM cars_drivers WHERE car_id = ?";
+        String newLinkQuery = "INSERT INTO cars_drivers (driver_id, car_id) VALUES (?, ?);";
```

Comment:
create 2 private methods, each opening its own connection, and call them after the `try-catch` block

Reply:
Created 2 private methods, calling them after `try-catch` bloc